agent/data/container_client.go

```diff
@@ -14,25 +14,70 @@
 package data
 
 import (
+	"encoding/json"
+
 	apicontainer "github.com/aws/amazon-ecs-agent/agent/api/container"
+	"github.com/aws/amazon-ecs-agent/agent/utils"
+
+	"github.com/boltdb/bolt"
 )
 
+// SaveDockerContainer saves a docker container to the container bucket.
+func (c *client) SaveDockerContainer(container *apicontainer.DockerContainer) error {
+	id := utils.GetTaskID(container.Container.TaskARN) + "-" + container.Container.Name
```

Comment:
Can we have a function for generating this id as this is used at multiple places,  this function can take taskArn and containerName as parameter and return the id string. 

Also, will it be a problem if a task has more than one container will same container name?

Reply:
added a function. for second point, i don't think containers in the same task can have same name